xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -252,6 +259,140 @@ public FeeResult fee() throws JsonRpcClientErrorException {
+   *        transaction will expire and won't be applied to the ledger.
+   * @param sequence The sequence number of the account submitting the {@link Transaction}. A {@link Transaction}
+   *        is only valid if the Sequence number is exactly 1 greater than the previous transaction
+   *        from the same account.
+   * @param account The unique {@link Address} of the account that initiated this transaction.
+   * @return {@code true} if the {@link Transaction} is final/validated else {@code false}.
+   */
+  public FinalStatus isFinal(
+    Hash256 transactionHash,
+    LedgerIndex submittedOnLedgerIndex,
+    UnsignedInteger lastLedgerSequence,
+    UnsignedInteger sequence,
+    Address account
+  ) {
+
+    return getValidatedTransaction(transactionHash)
+      .map(transactionResult -> {
+        //Do some stuff with transactionResult and return a FinalityStatus
+        String resultCode = transactionResult.metadata()
+          .orElseThrow(() -> new RuntimeException("Metadata not found in the validated transaction."))
+          .transactionResult();
+        if (resultCode.equals("tesSUCCESS")) {
+          return FinalStatus.VALIDATED_SUCCESS;
+        } else {
+          return FinalStatus.VALIDATED_FAILURE;
+        }
+      }).orElseGet(() -> {
+        try {
+          //Figure out if it's expired or that other account thing happened and return a FinalityStatus
+          if (FluentCompareTo.is(getMostRecentlyValidatedLedgerIndex()).lessThan(lastLedgerSequence)) {
+            // condition 2: last ledger seq hasn't passed yet
+            // possible solution would be to check again after a while to get the updated validation status.
+            return FinalStatus.NOT_FINAL;
+          } else {
+            // condition 3: last ledger seq has passed.
+            Range<UnsignedLong> submittedToLast = Range.closed(UnsignedLong.valueOf(submittedOnLedgerIndex.toString()),
+              UnsignedLong.valueOf(lastLedgerSequence.toString()));
+            boolean ledgerGapExists = this.serverInfo().completeLedgerRanges().stream()
+              .noneMatch(range -> range.encloses(submittedToLast));
+
+            if (ledgerGapExists) {
+              // wait to acquire the missing ledgers
+              // there is a possibility that the transaction with hash transactionHash exists on the missing ledgers
+              // possible solution would be to wait and query the missing ledgers once acquired, resubmit with
+              // updated fee and lastLedgerSequence if not found.
+              return FinalStatus.NOT_FINAL;
+            } else {
+              AccountInfoResult accountInfoResult = this.accountInfo(
+                AccountInfoRequestParams.of(account)
+              );
+              UnsignedInteger accountSequence = accountInfoResult.accountData().sequence();
+              if (FluentCompareTo.is(sequence).lessThan(accountSequence)) {
```

Comment:
I'm confused by this check - it seems like regardless of the current _account_ sequence, if the transaction doesn't exist on a ledger less than `lastLedgerSequence` (checked above on line 359) then it shouldn't matter what the account sequence is right?

If I'm wrong here, then we need to change RSubs, but the comment in RSubs is just this: 

```
// If we get here, it means the transaction was not found in the range of acceptable validated
// ledgers and never will be found, so expire the ReliableSubmission.
```

Reply:
This actually needs to be changed in RSubs I believe. If you look at [this flowchart](https://xrpl.org/img/reliable-tx-submission.svg), this check corresponds to the box that says (next to the yellow note):
```
Unknown transaction
has been validated.
Manual intervention
recommended
```

I'm not sure why we didn't implement this in RSubs, but I actually think we should wait until this gets merged and we should just delegate to this method in RSubs, instead of making a fix in the current RSubs method. This check is a super edge case and shouldn't really happen.